backend/src/survey_bot_v1.py

```diff
@@ -2,40 +2,64 @@
 from typing import List, Dict, Union, Tuple
 from check_objective import objective_met_agent
 from generate_question import question_generation_agent
+from request import Question
 
-class QuestionConfig(BaseModel):
-	followup_depth: int
-	criteria: List[str]
+import concurrent.futures
 
-class Question(BaseModel):
-	question: str
-	question_config: QuestionConfig
+PARALLEL_WORKERS = 4
+parallel_objective_met_agent_executor = concurrent.futures.ThreadPoolExecutor(max_workers=PARALLEL_WORKERS)
 
 class SurveyBotV1(BaseModel):
 	question_generation_agent = question_generation_agent
 	objective_met_agent = objective_met_agent
 	user_characteristics: Dict[str, Union[str, int, float, bool, None]] = {}
-	chat_history: List[List[Tuple[int, str, str]]] = []
+	# List of tuples of main question index, main question or followup question, user answer
+	chat_history: List[Tuple[int, str, str]] = []
+	# List of tuples of question and its left criteria
 	fixed_questions: List[Tuple[Question, List[str]]] = []
 	current_question_index: int = 0
 	current_question_followup_depth: int = 0
 
-	def __init__(self, fixed_questions):
-		self.fixed_questions = fixed_questions
+	def __init__(self, questions):
+		self.fixed_questions = [(question, question.question_config.criteria) for question in questions]
+
+	def parallel_objective_met_agent(self):
+		futures = [parallel_objective_met_agent_executor.submit(self.objective_met_agent, chat_history, \
+			fixed_questions[current_question_index].question, criteria) \
```

Comment:
`fixed_questions[current_question_index][0].question` will give the main question